### PR TITLE
fix cypress issues

### PR DESCRIPTION
**Pull request by cescoferraro**
This PR aims to fix several issue in the cypress testing 

## Changes & Fixes

- removes global testId variable
- makes createTest method return the testId
- fix promises issue with cypress that was passing failed tests
- start intecepting API requests
- remove a couple of unnecessary wait calls

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: web/cypress/integration/Home/EditTest.spec.ts [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
 
-    cy.get('[data-cy=edit-test-submit]').click();
-    cy.get('[data-cy=test-details-name]').should('have.text', `Edited Test (v2)`);
+      cy.get('[data-cy=edit-test-submit]').click();
+      cy.get('[data-cy=test-details-name]').should('have.text', `Edited Test (v2)`);
 
-    cy.location('pathname').should('match', /\/test\/.*/i, {timeout: 10000});
+      cy.location('pathname').should('match', testRunPageRegex, {timeout: 20000});
+      // this does not work here
+      // deleteTest();
```
Comment (mathnogueira) [inline web/cypress/integration/Home/EditTest.spec.ts:21]: any reason why it doesn't work? Shouldn't `deleteTest` receive an id instead of extracting it from the location?

File: web/package.json [hunk 2]
```diff
@@ -66,7 +66,7 @@
     "eject": "react-scripts eject",
     "cy:open": "cypress open",
     "cy:run": "cypress run",
-    "cy:ci": "cypress run --parallel --record --key $CYPRESS_RECORD_KEY",
+    "cy:ci": "cypress run --record --key $CYPRESS_RECORD_KEY",
```
Comment (mathnogueira) [inline web/package.json:69]: Why did you remove the `--parallel` flag?

Comment (cescoferraro) [reply web/package.json:69]: just testing

Comment (cescoferraro) [reply web/package.json:69]: right now i am isolating the stable tests from the one that fail all the time.
thats why I removed the parallel.

File: web/cypress/integration/Home/EditTest.spec.ts (hunk 1 above)
Comment (cescoferraro) [reply web/cypress/integration/Home/EditTest.spec.ts:21]: for some reason this deletion does not work. I have tried making deleteTest receive the testId.
but it fails the same way.

**Comment by cescoferraro** [comment]
Screen Shot 2022-07-20 at 15 19 05

the test inside the stable folder reliably pass.

**Comment by xoscar** [comment]
Seems like the changes done by @jorgeepc already fixed our e2e tests. Can we close this branch?

**Comment by jorgeepc** [comment]
> Seems like the changes done by @jorgeepc already fixed our e2e tests. Can we close this branch?

I believe @cescoferraro is trying to include some extra improvements. Not sure if they are ready.

File: web/cypress/integration/Home/EditTest.spec.ts [hunk 3]
```diff
@@ -1,43 +1,43 @@
...
-  });
-
-  beforeEach(() => {
-    cy.visit('http://localhost:3000/');
-  });
-
   it('should edit a test', () => {
-    cy.get(`[data-cy=test-actions-button-${testId}]`).click();
-    cy.get('[data-cy=test-card-edit]').click();
+    (async () => {
```
Comment (cescoferraro) [inline web/cypress/integration/Home/EditTest.spec.ts:5]: this is the fix of our issues. 
some-why tests were being marked as "passing" before the last promise returns.
those was the weird behaviour we were seeing

File: web/cypress/integration/Home/EditTest.spec.ts [hunk 4]
```diff
@@ -1,43 +1,43 @@
...
+      const testId = await createTest();
```
Comment (cescoferraro) [inline web/cypress/integration/Home/EditTest.spec.ts:6]: new createTest method that return an testId

File: web/cypress/integration/utils/Common.ts [hunk 5]
```diff
@@ -1,33 +1,69 @@
...
+
+export function inteceptHomeApiCall() {
+  cy.intercept({method: 'GET', url: '/api/tests'}).as('testList');
+}
+
+export function waitForTracePageApiCalls() {
+  cy.wait('@testRun');
+  cy.wait('@testObject');
+  cy.wait('@testRuns', {timeout: 20000});
+}
```
Comment (cescoferraro) [inline web/cypress/integration/utils/Common.ts:44]: we are waiting for api call instead of waiting x milliseconds. this makes tests more reliable

File: web/cypress/integration/Home/CreateTest.spec.ts [hunk 6]
```diff
@@ -24,7 +24,6 @@ describe('Create test', () => {
 
     cy.location('pathname').should('match', /\/test\/.*/i);
     cy.get('[data-cy=test-details-name]').should('have.text', `${name} (v1)`);
-    deleteTest();
```
Comment (jorgeepc) [inline web/cypress/integration/Home/CreateTest.spec.ts:27]: Are we deleting the tests in other place? We should delete them after each `test/spec`.

Comment (cescoferraro) [reply web/cypress/integration/Home/CreateTest.spec.ts:27]: I must have removed that when working on that. My bad, will fix it now

File: web/cypress.json [hunk 7]
```diff
@@ -1,8 +1,9 @@
 {
   "componentFolder": "cypress/components",
-  "testFiles": "**/*.spec.{js,ts,jsx,tsx}",
+  "testFiles": "{Home,Trace,TraceDetails}/**/*.spec.{js,ts,jsx,tsx}",
```
Comment (xoscar) [inline web/cypress.json:3]: Do we need this?

File: web/cypress.json [hunk 8]
```diff
@@ -1,8 +1,9 @@
...
   "viewportWidth": 1440,
   "viewportHeight": 1080,
   "responseTimeout": 30000,
+  "pageLoadTimeout": 60000,
```
Comment (xoscar) [inline web/cypress.json:7]: Is it really the page loading that slow?

**Review by xoscar — APPROVED** [review]
Hey @cescoferraro looks good, I left two minor non blocking comments
